### DISCUSSION TITLE
Required parameter model was null or undefined when calling createTranscription({...})

### DISCUSSION THREAD

**Discussion by sprihozhiy** [openai/whisper#1037] [Q&A] [answered]
Hey everyone,

I am struggling to understand what's wrong with the API call function (NodeJS).
Every time I try to call this way I am getting an error about the model parameter:

```
const options = {
  file:  fs.createReadStream('audio.mp3'),
  model: "whisper-1"
};

async function getTranscription() {
    const resp = await openai.createTranscription(options);
    console.log(resp.data);
}
```

but all good and works fine if I call this way:

```
async function getTranscription() {
  const resp = await openai.createTranslation(
        fs.createReadStream('audio.mp3'),
        "whisper-1"
    );
    console.log(resp.data);
 }
 ```
 
 What is wrong? Thanks!

**Comment by schnerd** [comment, marked as answer]
Hi there! Usually we prefer SDK-specific issues be added to the repo for that SDK – you can create node.js SDK issues in `openai-node` repo here: https://github.com/openai/openai-node/issues

The SDK auto-generator we are using currently only supports the second format for API endpoints that accept multipart form-data, so unfortunately we're stuck with that format for a little while. We display this in the example code on our website:

<img width="424" alt="image" src="https://user-images.githubusercontent.com/875591/223291701-bbc88f8b-4995-42cf-99ca-dc769c3bceb0.png">

We hope to improve this to be object-based in the future!

**Reply by sprihozhiy** [reply]
Hi David, thank you! I will create an issue in the repo you mentioned. But maybe you can guide me on how to set a `response_format` using the second format for API endpoints? I tried passing `responseFormat('srt')` as an argument but it didn't work. Thanks!

**Reply by schnerd** [reply]
You can find the actual SDK code here:
https://github.com/openai/openai-node/blob/0363de20747e272a92e41da4a4c4293104aa9461/api.ts#L2449

So you're final code might look something like

```js
async function getTranscription() {
  const resp = await openai.createTranslation(
        fs.createReadStream('audio.mp3'),
        "whisper-1",
        undefined, // prompt
        'srt', // responseFormat
    );
    console.log(resp.data);
 }
```
